godot-core/src/engine/file.rs

```diff
@@ -0,0 +1,766 @@
+    pub fn read_u8(&mut self) -> IoResult<u8> {
+        let val = self.fa.get_8();
+        self.check_error()?;
+        Ok(val)
+    }
+
+    /// Reads the next 16 bits from the file as [u16].
+    ///
+    /// For more information about underlying method see
+    /// [Godot documentation](https://docs.godotengine.org/en/stable/classes/class_fileaccess.html#class-fileaccess-method-get-16).
+    pub fn read_u16(&mut self) -> IoResult<u16> {
+        let val = self.fa.get_16();
+        self.check_error()?;
+        Ok(val)
+    }
+
+    /// Reads the next 32 bits from the file as [u32].
+    ///
+    /// For more information about underlying method see
+    /// [Godot documentation](https://docs.godotengine.org/en/stable/classes/class_fileaccess.html#class-fileaccess-method-get-32).
+    pub fn read_u32(&mut self) -> IoResult<u32> {
+        let val = self.fa.get_32();
+        self.check_error()?;
+        Ok(val)
+    }
+
+    /// Reads the next 64 bits from the file as [u64].
+    ///
+    /// For more information about underlying method see
+    /// [Godot documentation](https://docs.godotengine.org/en/stable/classes/class_fileaccess.html#class-fileaccess-method-get-64).
+    pub fn read_u64(&mut self) -> IoResult<u64> {
+        let val = self.fa.get_64();
+        self.check_error()?;
+        Ok(val)
+    }
+
+    /// Reads the Pascal String from file as [GodotString].
+    ///
+    /// Pascal String is useful for writing and retrieving verying-length string data from binary files.
+    /// It is saved with length-prefix, instead of null terminator as in C strings.
+    /// Text is interpreted as being UTF-8 encoded.
+    ///
+    /// See also:
+    /// - [Wikipedia article](https://en.wikipedia.org/wiki/String_(computer_science)#Length-prefixed)
+    /// - [Godot documentation](https://docs.godotengine.org/en/stable/classes/class_fileaccess.html#class-fileaccess-method-get-pascal-string)
+    pub fn read_pascal_string(&mut self) -> IoResult<GodotString> {
+        let val = self.fa.get_pascal_string();
+        self.check_error()?;
+        Ok(val)
+    }
+
+    /// Reads the whole file as [GodotString].
+    ///
+    /// If `skip_cr` is set to `true`, carriage returns will be ingored when parsing and only line
+    /// feed characters will represents a new line. Text is interpreted as being UTF-8 encoded.
+    ///
+    /// To retrieve the file as [String] instead, use [GFile::read_to_string()](https://doc.rust-lang.org/std/io/trait.Read.html#method.read_to_string)
+    /// [Read] trait method.
+    ///
+    /// For more information about underlying method see
+    /// [Godot documentation](https://docs.godotengine.org/en/stable/classes/class_fileaccess.html#class-fileaccess-method-get-as-text).
+    pub fn read_as_gstring_entire(&mut self, skip_cr: bool) -> IoResult<GodotString> {
+        let val = self.fa.get_as_text_ex().skip_cr(skip_cr).done();
+        self.check_error()?;
+        Ok(val)
+    }
```

Comment:
Just noticed, a `#[doc(alias = "get_as_text")]` might improve discoverability, as this method would then appear as a search result for `get_as_text` then.